packages/sdk/encrypted-backup/src/odis.ts

```diff
@@ -0,0 +1,260 @@
+    const odisOutputBase64 = await blindingClient.unblindAndVerifyMessage(
+      signatureResp.result.signature
+    )
+    odisOutput = Buffer.from(odisOutputBase64, 'base64')
+  } catch (error) {
+    return Err(new OdisServiceError(error as Error))
+  }
+
+  // Mix the key with the output from ODIS to get the hardened key.
+  return Ok(deriveKey(KDFInfo.ODIS_KEY_HARDENING, [key, odisOutput]))
+}
+
+/**
+ * Derive from the nonce a private key and use it to instanciate a wallet for request signing
+ *
+ * @remarks It is important that the auth key does not mix in entropy from the password value. If
+ * it did, then the derived address and signatures would act as a commitment to the underlying
+ * password value and would allow offline brute force attacks when combined with the other values
+ * mixed into the key value.
+ */
+export function odisQueryAuthorizer(nonce: Buffer): { address: Address; wallet: EIP712Wallet } {
+  // Derive the domain's request authorization key from the backup nonce.
+  const authKey = deriveKey(KDFInfo.ODIS_AUTH_KEY, [nonce])
+  const wallet = new LocalWallet()
+  wallet.addAccount(authKey.toString('hex'))
+  const address = wallet.getAccounts()[0]
+  if (address === undefined) {
+    // Throw the error instead if returning it as this is more akin to a panic.
+    throw new Error('Implementation error: LocalWallet with an added account returned no accounts')
+  }
+  return { address, wallet }
+}
+
+async function requestOdisQuotaStatus(
+  domain: SequentialDelayDomain,
+  environment: OdisServiceContext,
+  sessionID: string,
+  wallet?: EIP712Wallet
+): Promise<Result<DomainQuotaStatusResponseSuccess, BackupError>> {
+  const quotaStatusReq: DomainQuotaStatusRequest<SequentialDelayDomain> = {
+    domain,
+    options: {
+      signature: noString,
+      nonce: noNumber,
+    },
+    sessionID: defined(sessionID),
+  }
+
+  // If a query authorizer is defined in the domain, include a siganture over the request.
+  const authorizer = domain.address.defined ? domain.address.value : undefined
+  if (authorizer !== undefined) {
+    if (wallet === undefined || !wallet.hasAccount(authorizer)) {
+      return Err(
+        new AuthorizationError(
+          new Error('key for signing ODIS quota status request is unavailable')
+        )
+      )
+    }
+    quotaStatusReq.options.signature = defined(
+      await wallet.signTypedData(authorizer, domainQuotaStatusRequestEIP712(quotaStatusReq))
+    )
+  }
+
+  let quotaResp: DomainQuotaStatusResponse
+  try {
+    quotaResp = await queryOdis<DomainQuotaStatusResponse>(
+      { authenticationMethod: AuthenticationMethod.NONE },
+      quotaStatusReq,
+      environment,
+      Endpoints.DOMAIN_QUOTA_STATUS
+    )
+  } catch (error) {
+    if ((error as Error).message?.includes(ErrorMessages.ODIS_FETCH_ERROR)) {
+      return Err(new FetchError(error as Error))
+    }
+    if ((error as Error).message?.includes(ErrorMessages.ODIS_RATE_LIMIT_ERROR)) {
```

Comment:
We shouldn't ever receive a rate limit error on a domain status request, right?

Reply:
True. I've removed this check so it is reported as a generic unexpected error in 37a1137b6c640cbc64caa0b165431e331d047242